C-Sharp/sort/MergeSort.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Merge sort is an efficient, general-purpose, comparison-based sorting algorithm.
+// Implementation of merge sort
+
+using System;
+
+namespace MergeSortAlgorithm
+{
+    class Merge_Sort
+    {
+        //The merge step in the sorting problem to sort the elements
+        private static void Merge(int[] arr, int[] C, int l, int mid, int r)
```

Comment:
Instead of l and r use meaningful variable names. It doesn't change anything but makes code more readable.

Reply:
You have written the code so it makes sense to you but when you put it out to the public it is more readable if variable names make sense

Reply:
This whole merge function contains confusing variable names. Although it might work its a little confusing to me.

Reply:
Everything else looks great

Reply:
Thank you for your time and helpful review. I've fixed the variables names to more readable form.  I hope you can review my last commit.